File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/text_length.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeTextLength(statement: JSONObject, onError: (e: Exception) -> Unit): Int? {
    try {
        Log.d("hello", "execute_text_length")
        val pointer = statement["pointer"].toString()
        val text = executeStatement(pointer)?.toString()
        return text?.length
    } catch (e: Exception) {
        onError(e)
        return null
    }
}

File: piper/extensions/chrome/src/engine/statements/text_length.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let text = await this.executeStatement(stringify(pointer))
        return stringify(text).length
    } catch (e) {
        onError(e)
    }
    return null
}
